### DISCUSSION TITLE
Where can the Pin `MessageId` be found?

### DISCUSSION THREAD

**Discussion by TAINCER** [serenity-rs/serenity#2300] [Q&A]
When I send a new Message in a slash command, I want to remove the Pin Message after setting the Pin.
Since the [Pin](https://docs.rs/serenity/0.11.5/serenity/model/channel/struct.Message.html#method.pin) function doesn't return the `MessageId` for the Pin Message, I don't see how I can do this in the command function.

Example:
```rust
let test_message =
    ctx.send(|d| d.embed(|e| e.title("Test Message"))).await?;

let message = test_message.message().await?;

message.pin(ctx.http()).await?;
// Remove Pin Message?
```

**Comment by arqunis** [comment]
You can using the [message collector](https://serenity-rs.github.io/serenity/current/serenity/collector/message_collector/struct.CollectReply.html). Discord does not return data for the generated system message, since that's an action that is separate from pinning. Instead, you'll have to listen to incoming messages with the [PinAdd](https://serenity-rs.github.io/serenity/current/serenity/model/channel/enum.MessageType.html#variant.PinsAdd) message type:
```rust
use serenity::model::channel::MessageType;

let channel_id = message.channel_id;

if let Some(pin_msg) = channel_id.await_reply(ctx).filter(|m| m.kind == MessageType::PinAdd).await {
  pin_msg.delete(ctx).await;
}
```

**Reply by TAINCER** [reply]
How should I use this? If I `await_reply` after I pin the Message, the Collector won't catch the Pin Event. otherwise, if I first `await_reply` and then try to Pin the Message, the Collector will await until somewhere else a Pin is created.
I would like to put this in a separate Thread, but I can't clone the ctx.
